library/bignum.c

```diff
@@ -620,6 +647,12 @@ int mbedtls_mpi_read_file( mbedtls_mpi *X, int radix, FILE *fin )
      */
     char s[ MBEDTLS_MPI_RW_BUFFER_SIZE ];
 
+    MPI_VALIDATE_RET( X   != NULL );
+    MPI_VALIDATE_RET( fin != NULL );
+
+    if( radix < 2 || radix > 16 )
+        return( MBEDTLS_ERR_MPI_BAD_INPUT_DATA );
```

Comment:
Should we introduce new checks? It changes the library behaviour, and I believe we wanted to avoid that.

Reply:
I think this check is not actually new, it would be done later in `mbedtls_mpi_read_string()` anyway - so it never worked before with values outside that range. However I think it should be conditional.

Reply:
As mentioned above, I'm inclined to keep it.